### PR TITLE
HPCC-13207 Avoid uint overflow and crash if lookup HT huge.

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: thorlcr/activities/lookupjoin/thlookupjoinslave.cpp [hunk 1]
```diff
@@ -2183,7 +2183,10 @@ class CHTBase
     }
     void setup(CSlaveActivity *activity, rowidx_t size, IHash *_leftHash, IHash *_rightHash, ICompare *_compareLeftRight)
     {
-        size32_t sz = sizeof(const void *)*size;
+        unsigned __int64 _sz = sizeof(const void *)*size;
+        if (_sz > UINT_MAX) // treat as OOM exception for handling purposes.
+            throw MakeStringException(ROXIEMM_MEMORY_LIMIT_EXCEEDED, "Unsigned overflow, trying to allocate hash table of size: %" I64F "d ", _sz);
+        size32_t sz = _sz;
```
Comment (ghalliday) [inline thorlcr/activities/lookupjoin/thlookupjoinslave.cpp]: The memory manager supports > 2^32 byte allocations, so you should really be using a memsize_t here.  The check should be that the _sz = the cast value - which I think will almost always be true.
Also, should this be targetting 5.2?

**Comment by ghalliday** [comment]
@jakesmith I think this can be fixed in a better way

**Comment by jakesmith** [comment]
@ghalliday - please look again.

File: thorlcr/activities/lookupjoin/thlookupjoinslave.cpp [hunk 2]
```diff
@@ -2183,7 +2183,10 @@ class CHTBase
     }
     void setup(CSlaveActivity *activity, rowidx_t size, IHash *_leftHash, IHash *_rightHash, ICompare *_compareLeftRight)
     {
-        size32_t sz = sizeof(const void *)*size;
+        unsigned __int64 _sz = sizeof(const void *)*size;
```
Comment (ghalliday) [inline thorlcr/activities/lookupjoin/thlookupjoinslave.cpp]: Sorry Jake, going to be really picky...  If size_t is 32bit this multiply will be done as a 32bit operation - which means the test won't work.  At least one of the arguments needs to be cast to unsigned __int64 first!

**Comment by ghalliday** [comment]
@jakesmith one picky fix - otherwise the code looks fine.

**Comment by jakesmith** [comment]
@ghalliday - 3rd time lucky.

**Pull request merged**